File: app/src/main/java/com/example/amit/popularmovies/SearchActivity.java
package com.example.amit.popularmovies;

import android.app.LoaderManager;
import android.app.SearchManager;
import android.content.Context;
import android.content.CursorLoader;
import android.content.Intent;
import android.content.Loader;
import android.database.Cursor;
import android.net.Uri;
import android.os.Bundle;
import android.text.TextUtils;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ListView;
import android.widget.SimpleCursorAdapter;

import com.example.amit.popularmovies.data.MoviesContract;

import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.SearchView;
import androidx.appcompat.widget.Toolbar;

public class SearchActivity extends AppCompatActivity implements LoaderManager.LoaderCallbacks<Cursor>{


    private final static String SCREEN_LABEL = "Search";
    private SimpleCursorAdapter adapter;
    private int mPosition = ListView.INVALID_POSITION;


    SearchView mSearchView = null;
    String mQuery = "";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_result);
        Toolbar toolbar = (Toolbar)findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);

        String query = getIntent().getStringExtra(SearchManager.QUERY);
        query = query == null ? "" : query;
        mQuery = query;


        if (mSearchView != null) {
            mSearchView.setQuery(query, false);
        }

        overridePendingTransition(0, 0);

        handleIntent();
    }

    @Override
    protected void onNewIntent(Intent intent) {
        super.onNewIntent(intent);
        handleIntent();
    }
    private void handleIntent() {

    }
    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        super.onCreateOptionsMenu(menu);
        getMenuInflater().inflate(R.menu.search, menu);
        final MenuItem searchItem = menu.findItem(R.id.menu_search);
        if (searchItem != null) {
            SearchManager searchManager = (SearchManager) getSystemService(Context.SEARCH_SERVICE);
            final SearchView view = (SearchView) searchItem.getActionView();
            mSearchView = view;
            if (view == null) {

            } else {
                view.setSearchableInfo(searchManager.getSearchableInfo(getComponentName()));
                view.setIconified(false);
                view.setOnQueryTextListener(new SearchView.OnQueryTextListener() {
                    @Override
                    public boolean onQueryTextSubmit(String s) {
                        view.clearFocus();

                        return true;
                    }

                    @Override
                    public boolean onQueryTextChange(String s) {
                          fillData(s);
                        return true;
                    }
                });
                view.setOnCloseListener(new SearchView.OnCloseListener() {
                    @Override
                    public boolean onClose() {
                        finish();
                        return false;
                    }
                });
            }

            if (!TextUtils.isEmpty(mQuery)) {
                view.setQuery(mQuery, false);
            }
        }
        return true;
    }

    @Override
    public Loader<Cursor> onCreateLoader(int id, Bundle args) {
        String[] projection = {
                MoviesContract.MoviesEntry._ID,
                MoviesContract.MoviesEntry.COLUMN_MOVIE_TITLE,
                MoviesContract.MoviesEntry.COLUMN_MOVIE_POSTERPATH };
        CursorLoader cursorLoader = new CursorLoader(this,
                MoviesContract.MoviesEntry.CONTENT_URI, projection, null, null, null);
        return cursorLoader;
    }

    @Override
    public void onLoadFinished(Loader<Cursor> loader, Cursor data) {
        adapter.swapCursor(data);
    }

    @Override
    public void onLoaderReset(Loader<Cursor> loader) {
        adapter.swapCursor(null);
    }

    private void fillData(String q){
        ListView search_result_items =(ListView)findViewById(R.id.moviesearch_listView2);
        String[] COLUMNS_TO_BE_BOUND  = new String[] {
                MoviesContract.MoviesEntry.COLUMN_MOVIE_TITLE
        };

        int[] LAYOUT_ITEMS_TO_FILL = new int[] {
                android.R.id.text1
        };
        Uri search_db_query = MoviesContract.MoviesEntry.CONTENT_URI.buildUpon().build();
        String[] whereArgs = new String[] {
                q
        };
        getLoaderManager().initLoader(0, null, this);
        Cursor cursor=getContentResolver().query(search_db_query, null, "title LIKE '%"+q+"%'", null, null);
        adapter = new SimpleCursorAdapter(this,
                android.R.layout.simple_list_item_1,
                cursor,
                COLUMNS_TO_BE_BOUND,
                LAYOUT_ITEMS_TO_FILL,
                0);
        search_result_items.setAdapter(adapter);

        search_result_items.setOnItemClickListener(new AdapterView.OnItemClickListener() {

            @Override
            public void onItemClick(AdapterView<?> adapterView, View view, int position, long l) {

                Cursor cursor = (Cursor) adapterView.getItemAtPosition(position);
                Intent intent = new Intent(SearchActivity.this, DetailActivity.class)
                        .setData(MoviesContract.MoviesEntry.buildMoviesDetail(
                                cursor.getInt(0)));
                startActivity(intent);
                mPosition = position;
            }
        });
    }
}

File: app/src/main/java/com/amit/popularmovies/ui/search/SearchActivity.kt
package com.amit.popularmovies.ui.search

import android.app.LoaderManager
import android.app.SearchManager
import android.content.Context
import android.content.CursorLoader
import android.content.Intent
import android.content.Loader
import android.database.Cursor
import android.os.Bundle
import android.text.TextUtils
import android.view.Menu
import android.view.View
import android.widget.AdapterView.OnItemClickListener
import android.widget.ListView
import android.widget.SimpleCursorAdapter
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.SearchView
import androidx.appcompat.widget.Toolbar
import com.amit.popularmovies.R
import com.amit.popularmovies.ui.detail.DetailActivity

class SearchActivity : AppCompatActivity(){
//    private var adapter: SimpleCursorAdapter? = null
//    private var mPosition = ListView.INVALID_POSITION
//    var mSearchView: SearchView? = null
//    var mQuery = ""
//    override fun onCreate(savedInstanceState: Bundle?) {
//        super.onCreate(savedInstanceState)
//        setContentView(R.layout.activity_result)
//        val toolbar = findViewById<View>(R.id.toolbarView) as Toolbar
//        setSupportActionBar(toolbar)
//        supportActionBar!!.setDisplayHomeAsUpEnabled(true)
//        var query = intent.getStringExtra(SearchManager.QUERY)
//        query = query ?: ""
//        mQuery = query
//        if (mSearchView != null) {
//            mSearchView!!.setQuery(query, false)
//        }
//        overridePendingTransition(0, 0)
//        handleIntent()
//    }
//
//    override fun onNewIntent(intent: Intent) {
//        super.onNewIntent(intent)
//        handleIntent()
//    }
//
//    private fun handleIntent() {}
//    override fun onCreateOptionsMenu(menu: Menu): Boolean {
//        super.onCreateOptionsMenu(menu)
//        menuInflater.inflate(R.menu.search, menu)
//        val searchItem = menu.findItem(R.id.menu_search)
//        if (searchItem != null) {
//            val searchManager = getSystemService(Context.SEARCH_SERVICE) as SearchManager
//            val view = searchItem.actionView as SearchView
//            mSearchView = view
//            if (view == null) {
//            } else {
//                view.setSearchableInfo(searchManager.getSearchableInfo(componentName))
//                view.isIconified = false
//                view.setOnQueryTextListener(object : SearchView.OnQueryTextListener {
//                    override fun onQueryTextSubmit(s: String): Boolean {
//                        view.clearFocus()
//                        return true
//                    }
//
//                    override fun onQueryTextChange(s: String): Boolean {
//                        fillData(s)
//                        return true
//                    }
//                })
//                view.setOnCloseListener {
//                    finish()
//                    false
//                }
//            }
//            if (!TextUtils.isEmpty(mQuery)) {
//                view.setQuery(mQuery, false)
//            }
//        }
//        return true
//    }
//
//    override fun onLoadFinished(loader: Loader<Cursor>, data: Cursor) {
//        adapter!!.swapCursor(data)
//    }
//
//    override fun onLoaderReset(loader: Loader<Cursor>) {
//        adapter!!.swapCursor(null)
//    }
//
//    private fun fillData(q: String) {
//        val search_result_items = findViewById<View>(R.id.moviesearch_listView2) as ListView
//        val COLUMNS_TO_BE_BOUND = arrayOf(
//                MoviesContract.MoviesEntry.COLUMN_MOVIE_TITLE
//        )
//        val LAYOUT_ITEMS_TO_FILL = intArrayOf(
//                android.R.id.text1
//        )
//        val search_db_query = MoviesContract.MoviesEntry.CONTENT_URI.buildUpon().build()
//        val whereArgs = arrayOf(
//                q
//        )
//        loaderManager.initLoader(0, null, this)
//        val cursor = contentResolver.query(search_db_query, null, "title LIKE '%$q%'", null, null)
//        adapter = SimpleCursorAdapter(this,
//                android.R.layout.simple_list_item_1,
//                cursor,
//                COLUMNS_TO_BE_BOUND,
//                LAYOUT_ITEMS_TO_FILL,
//                0)
//        search_result_items.adapter = adapter
//        search_result_items.onItemClickListener = OnItemClickListener { adapterView, view, position, l ->
//            val cursor = adapterView.getItemAtPosition(position) as Cursor
//            val intent = Intent(this@SearchActivity, DetailActivity::class.java)
//                    .setData(MoviesContract.MoviesEntry.buildMoviesDetail(
//                            cursor.getInt(0)))
//            startActivity(intent)
//            mPosition = position
//        }
//    }
//
//    companion object {
//        private const val SCREEN_LABEL = "Search"
//    }
}